### PR TITLE
Rename python union types

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

This renames base classes to `BaseX` and removes the `Union` suffix from union aliases.

### CODE REVIEW

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
CI failing

File: bindings/python/iota_sdk/types/signature.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ class Ed25519Signature(Signature):
     type: int = field(default_factory=lambda: 0, init=False)
 
 
-SignatureUnion: TypeAlias = Union[Ed25519Signature]
+Signature: TypeAlias = Ed25519Signature
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/signature.py:24]: I don't know Python enough, wouldn't this be breaking change when we add more if it's not already a Union?

Comment (DaughterOfMars) [reply bindings/python/iota_sdk/types/signature.py:24]: No. And in fact my linter complained about using a union for a single variant

**Review by kwek20 — DISMISSED** [review]

File: bindings/python/iota_sdk/types/signature.py (hunk 1 above)
Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/signature.py:24]: Alright

**Review by thibault-martinez — APPROVED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
CI failing

File: bindings/python/tests/test_block.py [hunk 2]
```diff
@@ -20,7 +21,7 @@ def test_basic_block_with_tagged_data_payload():
             "data": "0x57652061726520616c6c206d616465206f662073746172647573742e0a436f756e743a20353436333730330a54696d657374616d703a20323032332d30372d31395430373a32323a32385a0a54697073656c656374696f6e3a20343732c2b573"}}
     block = BasicBlock.from_dict(block_dict)
     assert block.to_dict() == block_dict
-    assert isinstance(block.payload, Payload)
+    assert isinstance(block.payload, get_args(Payload))
```
Comment (thibault-martinez) [inline bindings/python/tests/test_block.py:24]: 😭

Comment (DaughterOfMars) [reply bindings/python/tests/test_block.py:24]: it's the only way

**Review by thibault-martinez — APPROVED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
CI failing

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
CI failing 😬

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]
Good stuff

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**